ray_on_golem/client/client.py

```diff
@@ -1,199 +1,23 @@
-import subprocess
-import time
-from datetime import datetime
-from functools import lru_cache
-from pathlib import Path
-from typing import Any, Dict, List, Optional, Tuple, Type, TypeVar, Union
+from typing import Any, Dict, List, Optional, Tuple, Type, TypeVar
 
 import requests
 from pydantic import BaseModel, ValidationError
 from ray.autoscaler._private.cli_logger import cli_logger  # noqa
 from yarl import URL
 
 from ray_on_golem.client.exceptions import RayOnGolemClientError, RayOnGolemClientValidationError
-from ray_on_golem.log import ZippingRotatingFileHandler
 from ray_on_golem.server import models, settings
-from ray_on_golem.server.models import CreateClusterResponseData, ShutdownState
-from ray_on_golem.server.settings import (
-    LOG_GROUP,
-    LOGGING_BACKUP_COUNT,
-    RAY_ON_GOLEM_CHECK_DEADLINE,
-    RAY_ON_GOLEM_PATH,
-    RAY_ON_GOLEM_SHUTDOWN_DEADLINE,
-    RAY_ON_GOLEM_START_DEADLINE,
-    get_log_path,
-)
-from ray_on_golem.utils import get_last_lines_from_file, is_running_on_golem_network
+from ray_on_golem.server.models import CreateClusterResponseData
 
 TResponseModel = TypeVar("TResponseModel")
 
 
 class RayOnGolemClient:
     def __init__(self, port: int) -> None:
-        self._base_url = URL("http://127.0.0.1").with_port(port)
-
+        self.port = port
```

Comment:
Is there any reason to have `port` exposed as a public field?

Reply:
yes, so that it can be taken from the client object instead of passing it around